File: yandex/ytcf/2018/final/a.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main () {
    string a, b, c;
    cin >> a >> b >> c;

    if (a.size() + b.size() != c.size()) {
        cout << "NO";
        return 0;
    }

    vector<int> dp(b.size() + 1);

    for (int i = 0; i <= a.size(); i ++)
        for (int j = 0; j <= b.size(); j ++)
            dp[j] = (i == 0 && j == 0) ||
                    (i > 0 && dp[j] && c[i + j - 1] == a[i - 1]) ||
                    (j > 0 && dp[j - 1] && c[i + j - 1] == b[j - 1]);

    cout << (dp[b.size()] ? "YES" : "NO");

    return 0;
}

File: yandex/ytcf/2018/final/a.py
a = input()
b = input()
c = input()

if len(a) + len(b) != len(c):
    print('NO')
    exit(0)

dp = [0] * (len(b) + 1)
for i in range(len(a) + 1):
    for j in range(len(b) + 1):
        dp[j] = i == 0 and j == 0 or \
                i > 0 and dp[j] and c[i + j - 1] == a[i - 1] or \
                j > 0 and dp[j - 1] and c[i + j - 1] == b[j - 1]

print('YES' if dp[len(b)] else 'NO')
